include/cuco/detail/trie/dynamic_bitset/kernels.cuh

```diff
@@ -0,0 +1,234 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuda/std/bit>
+
+namespace cuco {
+namespace experimental {
+namespace detail {
+
+/*
+ * @brief Gather bits of a range of keys
+ *
+ * @tparam BitvectorRef Bitvector reference type
+ * @tparam KeyIt Device-accessible iterator to input keys
+ * @tparam ValueIt Device-accessible iterator to values
+ * @tparam size_type Size type
+ *
+ * @param ref Bitvector ref
+ * @param keys Begin iterator to keys
+ * @param outputs Begin iterator to outputs
+ * @param num_keys Number of input keys
+ */
+template <typename BitvectorRef, typename KeyIt, typename ValueIt, typename size_type>
+__global__ void bitvector_get_kernel(BitvectorRef ref,
+                                     KeyIt keys,
+                                     ValueIt outputs,
+                                     size_type num_keys)
+{
+  uint32_t const loop_stride = gridDim.x * blockDim.x;
+  uint32_t key_id            = blockDim.x * blockIdx.x + threadIdx.x;
```

Comment:
```suggestion
  cuco::detail::index_type const loop_stride = gridDim.x * blockDim.x;
  cuco::detail::index_type key_id            = blockDim.x * blockIdx.x + threadIdx.x;
```
Similar to
https://github.com/NVIDIA/cuCollections/blob/c0cce866368396ced18a37a0d30287729d6af6c4/include/cuco/detail/common_kernels.cuh#L74-L75, please use `cuco::detail::index_type` consistently to avoid potential overflow issue. The same to other kernels.

Reply:
b26f326b6929531b6517305031e8c816402fd05a